contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/VoldemortBuildAndPushJob.java

```diff
@@ -816,6 +825,29 @@ public void runPushStore(Props props, String url, String dataDir) throws Excepti
 
         log.info("Push starting for cluster: " + url);
 
+        // CDN distcp
+        boolean cdnEnabled = props.getBoolean(PUSH_CDN_ENABLED, false);
+        String modifiedDataDir = new Path(dataDir).makeQualified(FileSystem.get(new JobConf())).toString();
+        String storeWhitelist = props.getString(PUSH_CDN_STORE_WHITELIST, null);
+        GobblinDistcpJob distcpJob = null;
+
+        if (cdnEnabled && storeWhitelist != null && storeWhitelist.contains(storeName)) {
+            try {
+                if (modifiedDataDir.matches(".*hdfs://.*:[0-9]{1,5}/.*")) {
+                    invokeHooks(BuildAndPushStatus.DISTCP_STARTING, url);
+                    distcpJob = new GobblinDistcpJob(getId(), modifiedDataDir, url, props);
+                    distcpJob.run();
+                    modifiedDataDir = distcpJob.getSource();
+                    invokeHooks(BuildAndPushStatus.DISTCP_FINISHED, url);
+                } else {
+                    warn("Invalid URL format! Skip Distcp.");
+                    throw new RuntimeException();
+                }
+            } catch (Exception e) {
+                invokeHooks(BuildAndPushStatus.DISTCP_FAILED, url);
```

Comment:
Sorry to interject, but although that seems useful on the surface, it kinds of goes against the current design of this abstraction.

There could be hooks which, for some reason, rely on this parameter being a proper URL. I'd rather not change the semantics of the hooks as part of this commit.